File: app/src/main/java/com/memoizlabs/kotlin_playground/currency/DefaultAndroidSchedulers.java
package com.memoizlabs.kotlin_playground.currency;

import rx.Scheduler;
import rx.android.schedulers.AndroidSchedulers;
import rx.schedulers.Schedulers;

public class DefaultAndroidSchedulers implements RxSchedulers {

    @Override
    public Scheduler getUiScheduler() {
        return AndroidSchedulers.mainThread();
    }

    @Override
    public Scheduler getIoScheduler() {
        return Schedulers.io();
    }
}

File: app/src/main/java/com/memoizlabs/kotlin_playground/currency/DefaultAndroidSchedulers.kt
package com.memoizlabs.kotlin_playground.currency

import rx.Scheduler
import rx.android.schedulers.AndroidSchedulers
import rx.schedulers.Schedulers

class DefaultAndroidSchedulers : RxSchedulers {
    override val uiScheduler: Scheduler = AndroidSchedulers.mainThread()
    override val ioScheduler: Scheduler = Schedulers.io()
}
